Assets/Scripts/Guests/Guest.cs

```diff
@@ -109,25 +161,134 @@ private void ChangeBehaviour()
 		}
 	}
 
+	/// <summary>
+	/// Finds the current room's offset that the guest is standing on
+	/// </summary>
+	/// <returns>The offset of the guest compared to the current room's pivot</returns>
+	private Vector2 FindCurrentOffset()
+	{
+		return transform.position - currentRoom.transform.position;
+	}
+
+	/// <summary>
+	/// Finds all the tiles that are one manhatten distance away from the guest
+	/// </summary>
+	/// <returns>All tiles neighbouring in the cardinal directions</returns>
+	private List<Vector2> FindPossibleTilesToMoveTo()
+	{
+		Vector2 currentOffset = FindCurrentOffset();
+		List<Vector2> possibleOffsets = new List<Vector2>();
+
+		foreach (Vector2 v in currentRoom.roomShape.OffsetFromRoomCenter)
+		{
+
+			if (Vector2.Distance(v, currentOffset) == 1f)
+			{
+				possibleOffsets.Add(v);
+			}
+		}
+
+		return possibleOffsets;
+	}
+
+	/// <summary>
+	/// Forces to leave the hotel
+	/// </summary>
+	public void ForceLeave()
+	{
+		enjoyTimeLeft = 0f;
+		numOfRoomsToVisit = 0;
+		BeginLeaving();
+
+	}
+
+
+	/// <summary>
+	/// Is the guest at the door?
+	/// </summary>
+	/// <returns>boolean that displays if the guest is on the door's position</returns>
+	private bool IsAtRoomDoor()
+	{
+		if (currentRoom == null) return true;

```

Comment:
what is a use case for this? is this defensive programming?

Reply:
The guest is currently in the Backrooms, and not inside a room